packages/mobile/src/store/offline-downloads/slice.ts

```diff
@@ -98,7 +106,11 @@ const slice = createSlice({
       state.downloadStatus[trackId] = OfflineDownloadStatus.ERROR
     },
     removeDownload: (state, { payload: trackId }: PayloadAction<string>) => {
-      delete state.downloadStatus[trackId]
+      if (state.downloadStatus[trackId] === OfflineDownloadStatus.ABANDONED) {
```

Comment:
So in this case delete doesn't fully delete status so we still have same denominator? When it it's tried to be removed again though we delete? Little confused!

Reply:
I didn't love this part a super ton, but basically we issue the same `removeDownload` calls from the worker erroring as well as when we actually try to delete/purge things. I figured that this was sorta ok. As in, the first pass removing puts things from ERROR to ABANDONED and then the second pass would actually delete them

:/ do you think this would be better as a separate action? I don't want to refactor a ton, but that might be better?

Reply:
All good! I agree it's a bit confusing but just wanted to understand intention for now

Reply:
i fix better